operators/src/processing/meteosat/temperature.rs

```diff
@@ -0,0 +1,678 @@
+    async fn test_ok() {
+        let ctx = MockExecutionContext::default();
+        let res = test_util::process(
+            || {
+                let props = test_util::create_properties(Some(4), Some(1), Some(0.0), Some(1.0));
+                let src = test_util::create_mock_source(props, None, None);
+
+                RasterOperator::boxed(Temperature {
+                    params: TemperatureParams::default(),
+                    sources: SingleRasterSource {
+                        raster: src.boxed(),
+                    },
+                })
+            },
+            test_util::create_mock_query(),
+            &ctx,
+        )
+        .await
+        .unwrap();
+
+        assert!(geoengine_datatypes::util::test::eq_with_no_data(
+            &res.grid_array,
+            &Grid2D::new(
+                [3, 2].into(),
+                vec![
+                    300.341_43,
+                    318.617_65,
+                    330.365_14,
+                    339.233_64,
+                    346.443_94,
+                    OUT_NO_DATA_VALUE,
+                ],
+                Some(OUT_NO_DATA_VALUE),
+            )
+            .unwrap()
+            .into()
+        ));
+    }
+
+    #[tokio::test]
+    async fn test_ok_force_satellite() {
+        let ctx = MockExecutionContext::default();
+        let res = test_util::process(
+            || {
+                let props = test_util::create_properties(Some(4), Some(1), Some(0.0), Some(1.0));
+                let src = test_util::create_mock_source(props, None, None);
+
+                RasterOperator::boxed(Temperature {
+                    params: TemperatureParams {
+                        force_satellite: Some(4),
+                    },
+                    sources: SingleRasterSource {
+                        raster: src.boxed(),
+                    },
+                })
+            },
+            test_util::create_mock_query(),
+            &ctx,
+        )
+        .await
+        .unwrap();
+
+        assert!(geoengine_datatypes::util::test::eq_with_no_data(
+            &res.grid_array,
+            &Grid2D::new(
+                [3, 2].into(),
+                vec![
+                    300.9428,
+                    319.250_15,
+                    331.019_04,
+                    339.9044,
+                    347.128_78,
+                    OUT_NO_DATA_VALUE
+                ],
+                Some(OUT_NO_DATA_VALUE),
+            )
+            .unwrap()
+            .into()
+        ));
+    }
+
+    // TODO: Cannot be tested, as the whole u8 range is a valid input. Think about making mock source u16
+    // #[tokio::test]
+    // async fn test_fail_illegal_input() {
+    //     let props = create_properties(Some(4), Some(1), Some(0.0), Some(1.0));
+    //     let mut params = TemperatureParams::default();
+    //     let res = process(props, params, None, Some(vec![1, 2, 3, 4, 1024, 0])).await;
+    //     assert!(res.is_err());
+    // }
```

Comment:
Could we use a generic for the mock source? But maybe this is a lot of work?…

Reply:
Yes, it is some amount of work. Maybe do it as a separate task?

Reply:
https://trello.com/c/xkolXr0t/340-mockrastersource-generischer-datentyp